pennylane/tape/qnode.py

```diff
@@ -779,6 +796,141 @@ def qfunc_decorator(func):
     return qfunc_decorator
 
 
+def _get_classical_jacobian(_qnode):
+    """Helper function to extract the Jacobian
+    matrix of the classical part of a QNode"""
+
+    def classical_preprocessing(*args, **kwargs):
+        """Returns the trainable gate parameters for
+        a given QNode input"""
+        _qnode.construct(args, kwargs)
+        return qml.math.stack(_qnode.qtape.get_parameters())
+
+    if _qnode.interface == "autograd":
+        return qml.jacobian(classical_preprocessing)
+
+    if _qnode.interface == "torch":
+        import torch
+
+        def _jacobian(*args, **kwargs):  # pylint: disable=unused-argument
+            return torch.autograd.functional.jacobian(classical_preprocessing, args)
+
+        return _jacobian
+
+    if _qnode.interface == "jax":
+        import jax
+
+        return jax.jacobian(classical_preprocessing)
+
+    if _qnode.interface == "tf":
+        import tensorflow as tf
+
+        def _jacobian(*args, **kwargs):
+            with tf.GradientTape() as tape:
+                tape.watch(args)
+                gate_params = classical_preprocessing(*args, **kwargs)
+
+            return tape.jacobian(gate_params, args)
+
+        return _jacobian
+
+
+def metric_tensor(_qnode, diag_approx=False, only_construct=False):
+    """metric_tensor(qnode, diag_approx=False)
+    Returns a function that evaluates the value of the metric tensor

```

Comment:
"evaluates the value" sounds awkward to me. Mabye "Returns a function that returns the metric tensor of a given QNode"

Reply:
```suggestion
    Returns a function that returns the value of the metric tensor
```